view.html.erb

```diff
@@ -1,6 +1,29 @@
-<form action="/rnode/<%= host %>/<%= port %>/login" method="post" target="_blank">
-  <input type="hidden" name="password" value="<%= password %>">
+<%
+  require 'digest'
+
+  # Generate form id, based on host and port
+  form_id = Digest::SHA1.hexdigest("--" + host.to_s + "--" + port.to_s + "--")
+
+  # Generate SHA256 digest of Code Server Password
+  cookieValue = Digest::SHA256.hexdigest(password)
+-%>
+
+<script>
+function setCookie(cname, cvalue, exdays) {
+  var d = new Date();
+  d.setTime(d.getTime() + (exdays*24*60*60*1000));
+  var expires = "expires="+ d.toUTCString();
+  document.cookie = cname + "=" + cvalue + ";" + expires + ";path=/rnode/" + "<%= host.to_s %>" + "/" + "<%= port.to_s %>/";
+}
+
+var s = document.getElementById("<%= form_id %>").closest(".panel").id;
+var p = "<%= cookieValue %>";
+var c = "key";
+var e = 7;
```

Comment:
This should be placed in an anonymous function. Right now s, p, c, e are all global variables. so what happens when there are two sessions at the same time?

```js
<script>
(function(){

function setCookie(...

})();
</script>
```

Reply:
Weird behavior happens. Sometimes `N` amount of code-server sessions work, sometimes they don't.

I don't think an anonymous function like your example is good here, what if we had an event listener on the form button click and set the cookie inside of there?

Reply:
Using jQuery I think the solution would look something like this:

```html
<script>
(function(){
  var setCookie = function(cookieName, cookieValue, expiration) {
    // impl
  };

  $("#<%= form_id %>").submit(function(){
    setCookie("key", "<%= cookieValue %>", 7);
  });
})();
</script>

<form id="<%= form_id %>" action="/rnode/<%= host %>/<%= port %>/" method="GET" target="_blank">
    <button class="btn btn-primary" type="submit" id="<%= form_id %>_button">
      <i class="fa fa-cogs"></i> Connect to VS Code
    </button>
</form>
```

Looking at the code under review it looks like `var s = document.getElementById("<%= form_id %>").closest(".panel").id;` is not used...